core/src/main/scala/kafka/server/metadata/BrokerMetadataSnapshotter.scala

```diff
@@ -109,4 +115,9 @@ class BrokerMetadataSnapshotter(
     beginShutdown()
     eventQueue.close()
   }
+
+  // VisibleForTesting
+  def currentSnapshotOffset(): Long = {
+    _currentSnapshotOffset
+  }
```

Comment:
It doesn't look like you use this in any of the tests. Let's remove this method if true.